packages/kit/src/runtime/client/renderer.js

```diff
@@ -39,6 +40,44 @@ function notifiable_store(value) {
 	return { notify, set, subscribe };
 }
 
+/**
+ * @param {any} value
+ * @param {(set: (new_value: any) => void) => any} fn
+ * @param {number | undefined} interval
+ */
+function checkable_store(value, fn, interval) {
+	const { set, update, subscribe } = writable(value);
+
+	setInterval(() => {
+		fn(set);
+	}, interval);
+
+	return {
+		set,
+		update,
+		subscribe,
+		check: () => fn(set)
+	};
+}
+
+/**
+ * @returns {Promise<boolean>}
+ */
+async function has_version_changed() {
+	if (import.meta.env.DEV) return false;
+
+	const headers = new Headers();
+	headers.append('pragma', 'no-cache');
+	headers.append('cache-control', 'no-cache');
+
+	const current_version = import.meta.env.VITE_APP_VERSION;
+	const new_version = await fetch('_app/version.json', { headers })
```

Comment:
This needs to use `config.kit.appDir` somehow, it can't be hardcoded to `_app`.

Also, no need to create a `Headers` object, you can just do this:

```js
const new_version - await fetch(`${base}/${app_dir}/version.json`, {
  headers: {
    'pragma': 'no-cache',
    'cache-control': 'no-cache'
  }
})
```

We shouldn't be swallowing the error here — at the very least an explicit call to `await updated.check()` should throw if the fetch fails.

Reply:
I added that catch because of error messages in the console in dev. Since dev is handled differently now, this is not necessary anymore.